### DISCUSSION TITLE
Insert new addresses to self-hosted instance

### DISCUSSION THREAD

**Discussion by Dimii96** [osm-search/Nominatim#3551] [Q&A]
Hello,

I have a self-hosted instance of nominatim. There are scenarios where the OSM data will not have addresses that a user will look up.

Are we able to insert new address info in my own database instance? I only use my instance for address look ups and do not have map interactions, so only need data like: number, street, postcode, suburb, state, city, lat, long (don't need node data etc) 

I am aware I can update the OSM database and do a db update on my instance with the changes, but I can't quite rely on this due to inefficiency of requiring to do an update each time. This is because I would like to have new addresses available on insert.

(I do have a plan to have the new addresses added to OSM db every so often)

Thanks in advance

**Comment by lonvia** [comment]
You can add additional data using `nominatim add-data`. The data has to be in OSM format and for addresses, there has to be an object for the street in the database already or things will not work properly.

See also  my [answer in the forum](https://community.openstreetmap.org/t/need-help-importing-custom-data-into-nominatim/118952/5) on that topic.
